giraffe/README.md

```diff
@@ -0,0 +1,341 @@
+
+- **onSetYDomain**: _function(array[min, max]). Optional._ See above regarding **yDomain**.
+
+- **onResetYDomain**: _function(). Optional._ See above regarding **yDomain**.
+
+### Legend Tooltip properties
+
+- **legendFont**: _string. Optional._ The [_CSS font_](https://developer.mozilla.org/en-US/docs/Web/CSS/font) value for the styling of the legend (tooltip).
+
+- **legendFontColor**: _string. Optional._ The _CSS color value_ of the column headers in the legend (tooltip). The rest of the legend will use the color scheme set by the `LayerConfig`'s `colors` options.
+
+- **legendFontBrightColor**: _string. Optional. Recommendation: do not include._ The _CSS color value_ of any text that is not a column header or in a row inside the legend (tooltip). Currently no such text exists and may be added in the future.
+
+- **legendBackgroundColor**: _string. Optional._ The _CSS color value_ of the background in the legend (tooltip).
+
+- **legendBorder**: _string. Optional._ The [_CSS border_](https://developer.mozilla.org/en-US/docs/Web/CSS/border) value for the styling of the border around the legend (tooltip).
+
+- **legendCrosshairColor**: _string | Object. Optional._ The _CSS color value_ or styling of the vertical crosshair line through the Plot at where the mouse is hovering, defined as a [_CanvasRenderingContext2D strokeStyle_](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/strokeStyle).
+
+- **legendColumns**: _array[string, ...]. Optional._ When included, this array will determine which column key names that should be included in the legend (tooltip). If this option is included as an empty array, the legend will be empty.
+
+## Utility Functions
+
+Giraffe comes with utility functions.
+
+- **fromFlux**: _function(string)._ Takes a Flux CSV, converts it, and returns a `Table` used in the [table property](#data-properties) of the config.
+
+- **fromRows**: _function([Object, ...], Object)._ The first argument is an array of objects, each representing a row of data. The optional second argument describes the schema for the data. Returns a `Table` used in the [table property](#data-properties) of the config.
+
+- **newTable**: _function(number)._ The argument is a length for a newly created `Table` with no initial data that allows only columns equal to that length to be added. Returns the created `Table`.
+
+## LayerConfig
+
+- **LineLayerConfig**: _Object_. Maximum one per `<Plot>`. Properties are:
+
+  - **type**: _"line". **Required**_. Specifies that this LayerConfig and `<Plot>` is a line graph.
```

Comment:
Isn't the type a `string` that can be an enum value of `line`?

Reply:
It's very hard to be clear in docs without writing real code. This is pseudo-code to say that the property name is "type" and the value is literally called "line". Not aiming to bring TypeScript semantics into the docs.

Reply:
Totally. I was still waking up when I was reviewing this and didn't connect that this was a subset of `LineLayerConfig` 🤦